crates/sequencer/src/sequencer.rs

```diff
@@ -25,3 +36,173 @@ where
         Self { context, state }
     }
 }
+
+impl<S> Execution for Sequencer<S>
+where
+    for<'a> &'a mut S: State + StateReader + Committer<S>,
+{
+    fn execute(&mut self, transaction: Transaction) -> Result<(), TransactionExecutionError> {
+        let mut cached_state = CachedState::new(&mut self.state);
+        let res = transaction.execute(&mut cached_state, &self.context, false);
+
+        match res {
+            Err(err) => {
+                warn!("Transaction execution failed: {:?}", err)
+            }
+            Ok(execution_information) => {
+                <&mut S>::commit(&mut cached_state);
+                match execution_information.revert_error {
+                    Some(err) => {
+                        warn!("Transaction execution failed: {:?}", err)
+                    }
+                    None => {
+                        trace!("Transaction execution succeeded {execution_information:?}")
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::fs::File;
+    use std::sync::Arc;
+
+    use blockifier::abi::abi_utils::get_storage_var_address;
+    use blockifier::execution::contract_class::{ContractClass, ContractClassV0};
+    use blockifier::state::state_api::State as BlockifierState;
+    use blockifier::transaction::account_transaction::AccountTransaction;
+    use starknet::core::types::FieldElement;
+    use starknet_api::core::{ChainId, ClassHash, ContractAddress, Nonce};
+    use starknet_api::hash::StarkFelt;
+    use starknet_api::transaction::{
+        Calldata, Fee, InvokeTransaction, InvokeTransactionV1, TransactionHash,
+        TransactionSignature,
+    };
+
+    use crate::constants::test_constants::{
+        FEE_TOKEN_ADDRESS, ONE_BLOCK_NUMBER, ONE_BLOCK_TIMESTAMP, ONE_CLASS_HASH,
+        SEQUENCER_ADDRESS, TEST_ADDRESS, TEST_CONTRACT_ACCOUNT, TEST_CONTRACT_ADDRESS,
+        TWO_CLASS_HASH, ZERO_FELT,
+    };
+    use crate::state::State;
+
+    use super::*;
+
+    fn read_contract_class(path: &str) -> ContractClass {
```

Comment:
```suggestion
    fn read_contract_class_v0(path: &str) -> ContractClass {
```

Reply:
could we add some v1 contracts as well?

Reply:
I'm going to check but first I need to compile a cairo 1 program into a json format, not sure if this is possible rn

Reply:
ok to do in another issue/pr

Reply:
Done!